packages/grid/_modules_/grid/hooks/features/rows/useGridRows.ts

```diff
@@ -54,168 +56,202 @@ export function convertGridRowsPropToState(
  */
 export const useGridRows = (
   apiRef: GridApiRef,
-  props: Pick<GridComponentProps, 'rows' | 'getRowId' | 'rowCount'>,
+  props: Pick<GridComponentProps, 'rows' | 'getRowId' | 'rowCount' | 'throttleRowsMs'>,
 ): void => {
   const logger = useGridLogger(apiRef, 'useGridRows');
   const [, setGridState, forceUpdate] = useGridState(apiRef);
-  const stateRows = useGridSelector(apiRef, gridRowsStateSelector);
-  const updateTimeout = React.useRef<any>();
-
-  const delayedForceUpdate = React.useCallback(
-    (preUpdateCallback?: Function) => {
-      if (updateTimeout.current == null) {
-        updateTimeout.current = setTimeout(() => {
-          logger.debug(`Updating component`);
-          updateTimeout.current = null;
-          if (preUpdateCallback) {
-            preUpdateCallback();
-          }
-          forceUpdate();
-        }, 100);
-      }
-    },
-    [logger, forceUpdate],
-  );
 
-  const internalRowsState = React.useRef<InternalGridRowsState>(stateRows);
+  const rowsCache = React.useRef() as React.MutableRefObject<GridRowsInternalCache>;
```

Comment:
It was to avoid calling  `getInitialGridRowState` on every render

Otherwise I can just do

```tsx
  const rowsCache = React.useRef<GridRowsInternalCache>({
      state: getInitialGridRowState(),
      timeout: null,
      lastUpdateMs: null,
  });
```

But anyway, with the synchronous state PR, the state will be initialized before this hook first call
We will be able to do

```tsx
  const rowsCache = React.useRef<GridRowsInternalCache>({
      state: apiRef.current.state.rows,
      timeout: null,
      lastUpdateMs: null,
  });
```

So the solution is temporary.

Reply:
If you call `getInitialGridRowState` outside the hook, then it won't be called on every render.

Reply:
I supposed Damien created callback for initial state value instead of objects to avoid having the same object reference across Grid instances or various calls.

In this situation I don't think it matters a lot, but that's why I avoided calling `getInitialGridRowState` outside of the hook.

Reply:
If it's temporary, we can keep it.